besspin/cyberPhys/cyberphyslib/cyberphyslib/kiosk/kiosk.py

```diff
@@ -489,77 +563,149 @@ def hackActive(self, hack_id) -> bool:
         else:
             return True
 
+    def buttonPressed(self, button_id) -> bool:
+        """
+        Mimic an infotainment client and send button press to
+        the hacked info server, over UDP CAN network with a special port
+        """
+        print(f"CAN_ID_BUTTON_PRESSED: {ComponentDictionary[button_id]}")
+        if self.deploy_mode:
+            try:
+                msg = Message(arbitration_id=canlib.CAN_ID_BUTTON_PRESSED,
+                            data=struct.pack(canlib.CAN_FORMAT_BUTTON_PRESSED, button_id))
+                self.infotainment_bus.send(msg)
+                return True
+            except Exception as exc:
+                print(f"<{self.__class__.__name__}> Error sending message: {msg}: {exc}")
+                return False
+        else:
+            return True
+
     def execute_infotainment_hack(self, arg):
+        """
+        Parse `arg`, and either:
+        * send `buttonPressed` message (volume/music)
+        * or listen to the incoming position messages (GPS exfil)
+        * updates `retval` accordingly
+        """
         self.hackActive(canlib.HACK_INFOTAINMENT_1)
-        # TODO if self.deploy_mode:
         if arg == "volumeUp":
-            # TODO: proper send
+            self.buttonPressed(canlib.BUTTON_VOLUME_UP)
             self.ipc_msg['retval'] = "Volume increased"
         elif arg == "volumeDown":
-            # TODO: proper send
+            self.buttonPressed(canlib.BUTTON_VOLUME_DOWN)
             self.ipc_msg['retval'] = "Volume decreased"
         elif arg == "exfil":
-            # TODO: proper send
-            self.ipc_msg['retval'] = "124 W, 11N"
+            # TODO: we need to listen to the CAN_ID_CAR_X/Y/Z/R from the hacked server
```

Comment:
@dmzimmerman What do you recommend here? Just show the latest X/Y/Z coordinates we grab?

Reply:
Yup, that'd be my recommendation. "grab" is a bit of a misnomer, of course, because they're pushed, not pulled.